### DISCUSSION TITLE
Steinel PIR not refreshed

### DISCUSSION THREAD

**Discussion by hestiap** [zwave-js/zwave-js-ui#2043] [Q&A] [answered]
Hi,
I have a PIR sensor with relay (Steinel IS140-2) that is not refreshed all the time.
So when we go in front of the sensor, nothing happened in zwave-js.
If I click refresh it works
![image](https://user-images.githubusercontent.com/73014953/145670560-ecd2f94c-2b38-4632-980c-ef7193cfc652.png)
How to fix this issue?
I've already made several Heal, Re-Interview, Refresh: perhaps there is an order to do them?
```
zwavejs2mqtt: 5.12.0
zwave-js: 8.7.6
home id: 3795207774
home hex: 0xe236465e
```
![image](https://user-images.githubusercontent.com/73014953/145670613-bd72c9de-3972-4e28-ab9c-2717049c8405.png)

**Comment by robertsLando** [comment]
@hestiap You should check your group association (check if lifeline association is correctly setted up) and device configuration CC to see if reporting is enabled. 

cc @AlCalzone

**Reply by AlCalzone** [reply]
@hestiap 
Please make a [driver log](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs), loglevel `debug` or `silly` and attach it here as a file (using drag&drop).
While the log is running, re-interview the device, wait until that is done and trigger it.

**Reply by hestiap** [reply]
@AlCalzone 
[Uploading zwavejs2mqtt_2021-12-25.log…]()
here is a log, but it didn't succeed in the re-interviews of the both motion sensor, IDs 80 and 99
I'll try again

**Comment by hestiap** [comment]
thanks to both of you
I'll do that...
At the present time, I'm stuck because this morning I've restarted the server and after several hours no devices were with a good status. 
I've tried to restart the container, to re-interview some nodes... Nothing, no progress
![image](https://user-images.githubusercontent.com/73014953/145861627-d72342e8-ef6c-4423-ad2e-70b49ef98c55.png)
![image](https://user-images.githubusercontent.com/73014953/145861696-fad1187b-c4ed-4822-87dd-e689b4bf3274.png)
What could I do to "unlock" the system?
Thanks again for your help

**Comment by hestiap** [comment]
After some actions: restart server, restart zwavejs2mqtt, cut the power of the devices... the situation went back to "normal".
I've tried a note re-interview on another PIR (same model) yesterday and it's also stuck on NodeInfo
![image](https://user-images.githubusercontent.com/73014953/145949417-89ed4b46-9902-4f0a-8eab-b5950119f6b6.png)
But it's reporting lux anyway.
How to get it back to "normal"?
zwavejs2mqtt: 5.12.0
zwave-js: 8.7.6

**Reply by AlCalzone** [reply]
> How to get it back to "normal"?

I guess you have some connectivity problems. Open an issue in https://github.com/zwave-js/node-zwave-js, include the requested logs (including a startup or re-interview of the node), then I can take a look.

**Comment by robertsLando** [comment]
@hestiap Did you also tried with latest version? 6.1.1

**Reply by hestiap** [reply]
I will...
![image](https://user-images.githubusercontent.com/73014953/145992667-a2970cb4-89b6-4b7b-b6a1-048e5a4a0b4f.png)
It seems it is not possible to upgrade online
I'm not very expert with containers...
I'm going to look at it

**Reply by robertsLando** [reply]
@hestiap That's used to update config db not the application itself. You need to update the application using docker or whatelse you used to run it

**Reply by hestiap** [reply]
I've set up a env test with a backup.
I've upgraded to zwavejs2mqtt: 6.1.0 / zwave-js: 8.9.0-beta.0
The nodes are being re-interviewing...
If ok, I'll upgrade the prod

**Reply by robertsLando** [reply]
It should be safe, bBTW now there is also beta 2, I would wait for zwavejs2mqtt 6.2.0 to be sure

**Comment by hestiap** [comment, marked as answer]
@robertsLando 
I didn't see your reply in time!
I've ugraded the prod to zwavejs2mqtt: 6.1.1 / zwave-js: 8.9.0-beta.1
The device went back to life :-)
I don't know if it's relative to the new version or to another stop/start + re-interview
Let's see if it keeps "alive" for a long time!

**Comment by hestiap** [comment]
@robertsLando 
After a while the device didn't report again, so no movement detection :-(
I had to refresh it... so for me the whole solution it's not very stable.
Are there a way (automatic) to know if a device no longer reports and to refresh it?

**Reply by robertsLando** [reply]
> Are there a way (automatic) to know if a device no longer reports and to refresh it?

Nope but you could enable poll of that specific value

**Reply by hestiap** [reply]
I have issue on several device, I could not poll all of them.
Theses ones are motion sensors so I need real-time information.
So the point is to get them more reliable. Yesterday there were no longer reporting

**Comment by hestiap** [comment]
@robertsLando 
@AlCalzone 
I though I have answered but could not find my reply ?!? My mistake
I have a log with the previous version when re-interviewing, but with no success (zwavejs2mqtt_2021-12-25.log)
I've upgraded to 6.2.0 and set the log at restart (zwavejs2mqtt_2021-12-26.log).
Not better for the end of the re-interview
![image](https://user-images.githubusercontent.com/73014953/147421801-8d153889-34db-4e5b-8d00-8edf8a5a9a78.png)
[zwavejs2mqtt_2021-12-25.log](https://github.com/zwave-js/zwavejs2mqtt/files/7777701/zwavejs2mqtt_2021-12-25.log)
[zwavejs2mqtt_2021-12-26.log](https://github.com/zwave-js/zwavejs2mqtt/files/7777702/zwavejs2mqtt_2021-12-26.log)

**Reply by robertsLando** [reply]
Seems your device is stucked on NodeInfo stage

**Reply by AlCalzone** [reply]
@hestiap I need to see [driver logs](https://zwave-js.github.io/zwavejs2mqtt/#/troubleshooting/generating-logs?id=driver-logs), on loglevel `debug`. The application logs don't help diagnose this.

**Reply by hestiap** [reply]
@AlCalzone 
sorry!
hopefully they were on
[zwavejs_2021-12-27.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7779959/zwavejs_2021-12-27.zip)
[zwavejs_2021-12-26.log.gz](https://github.com/zwave-js/zwavejs2mqtt/files/7779977/zwavejs_2021-12-26.log.gz)

**Comment by hestiap** [comment]
@AlCalzone 
I cut the power and re-interviewed both devices and the interviews finished in some seconds.
The log were still on but no file were written (previously I've moved the log to tmp...).
I set the log off and on and re-interview  one (80)
The logs
[zwavejs_2021-12-27b.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7780130/zwavejs_2021-12-27b.zip)

**Reply by AlCalzone** [reply]
Ok that log looks okay and the lifelin association is set up. Can you make another one (without re-interviewing) where you trigger the sensor so I can see what gets sent?

**Reply by hestiap** [reply]
thanks!
here it is...
[zwavejs_2021-12-27.zip](https://github.com/zwave-js/zwavejs2mqtt/files/7781243/zwavejs_2021-12-27.zip)
2 nodes with the same devices: 80 and 99

**Reply by AlCalzone** [reply]
Seems okay now, doesn't it?
```
2021-12-27T17:49:29.385Z CNTRLR   [Node 080] [~] [Multilevel Sensor] Illuminance: 14 => 13          [Endpoint 0]
2021-12-27T17:50:17.722Z DRIVER « [Node 080] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   Home Security
                                      notification status: 255
                                      notification state:  Motion detection
2021-12-27T17:50:17.848Z CNTRLR   [Node 080] [~] [Binary Switch] currentValue: false => true        [Endpoint 0]
2021-12-27T17:50:29.986Z DRIVER « [Node 080] [REQ] [ApplicationCommand]
                                  └─[NotificationCCReport]
                                      notification type:   7
                                      notification status: 255
                                      notification state:  idle
                                      event parameters:    0x08
2021-12-27T17:50:54.923Z CNTRLR   [Node 080] [~] [Binary Switch] currentValue: true => false        [Endpoint 0]
```
Node 099 looks similar.

**Reply by hestiap** [reply]
yes, now...
but tomorrow
sometimes they do not report motion, nor illuminance. They stay ok only a few day :-(
Right now the "same" for the node 54, stuck on 'On' for a motion sensor. I had to refresh it
Regarding PIRs I consider it is not reliable enough.
Is there any way to know if the node is "stuck"? And to refresh it?

**Reply by AlCalzone** [reply]
> Is there any way to know if the node is "stuck"? And to refresh it?

You could probably build an automation that refreshes the values if they didn't change in a while.

However that seems to be either a connectivity issues or an issue with the device itself. Refreshing the values merely masks the symptoms.

**Reply by hestiap** [reply]
> However that seems to be either a connectivity issues or an issue with the device itself. Refreshing the values merely masks the symptoms.

The first could be to monitor if there is a connectivity issue in order to fix it later
At the present time, it is just when the node is needed that the issue is known
I think there is an issue at the starting because when I restart zjs2mqtt some devices are lost...

> build an automation that refreshes the values if they didn't change in a while.

Could you give more direction please?

**Reply by AlCalzone** [reply]
> The first could be to monitor if there is a connectivity issue

That's not easy to diagnose "live" though.

> Could you give more direction please?

If you're using HomeAssistant, they have community support for such automations.
In standalone zwavejs2mqtt, maybe @robertsLando has some ideas.
Otherwise, I'm of not much help.

**Reply by robertsLando** [reply]
@hestiap The easier way could be to build it using node-red, simply use an mqtt in node to receive the value update (usiong the topic of that value), set a timeout when you receive the value and clear any previous timeout, when the timeout triggers send a mqtt packet using mqtt out node that triggers the refreshValues api against that node

**Reply by hestiap** [reply]
@robertsLando 
thanks your hint and sorry for my late reply
I'm just beginning with Node-RED, I think I could do it. I could also do it from domoticz, or both...
Could you give me the link to the refreshValues API pls? I didn't find it yesterday
With this, it's not covering the today's "issue"...
I have a switch that switched on the light after I cut and restarted the power, but didn't refresh zjs2mqtt
So it stayed 'On' for a long time :-(
and this switch doesn't report the power so it's usually mute!

**Reply by robertsLando** [reply]
@hestiap You have to call refreshValuesCC api using nodeid and commandClass number as params. In order to send that api request using mqtt follow steps here: https://zwave-js.github.io/zwavejs2mqtt/#/guide/mqtt?id=api-call-examples

**Comment by hestiap** [comment]
@robertsLando 
thanks for the info about the API
Before eventually setting up an automatic refresh, I'd like to monitor the nodes that doesn't have given info for a "long" time.
I've tried
Node status: <mqtt_prefix>/<?node_location>/<node_name>/status and Node information: <mqtt_prefix>/<?node_location>/<node_name>/nodeinfo
with no success:
zwave/147/status
zwave/testloc/testname/status
zwave/testname/status

![image](https://user-images.githubusercontent.com/73014953/152007758-902eb3c5-7748-4a50-b111-f570e65b1448.png)
![image](https://user-images.githubusercontent.com/73014953/152007461-509b1a84-acf3-4cab-81e3-e9ad280ab366.png)
Nothing visible on zwavzjs2mqtt
Nothing back visible on the broker
Sorry for noob's questions, but most of this is new for me!

**Reply by robertsLando** [reply]
@hestiap Why are you publishing to status topic?

**Comment by hestiap** [comment]
@robertsLando 
That's a try
I think I need nodeinfo because there is lastActive
But it is the same with zwave/147/nodeinfo
I've not managed to make it work

**Reply by robertsLando** [reply]
@hestiap yeah but I don't understand why you are publishing a message to that topic, mqtt is an event based protocol, you just need to subscribe to a topic to get informations. Publish to send actions or reports (if you are a sensor, in your case you use publish just to send actions)

**Comment by hestiap** [comment]
@robertsLando 
I've understood your point.
My point is the way I think I could do is to request the information at a time and to compare it to the lastActive. If lastActive is too old, an alarm is risen
 If I subscribe to the nodeinfo topic to get lastActive, when I receive it, it will be at "the time", no way to compare it! So I'll have to store the information to check later if the node was not buzy.
But perhaps I miss something and there is another way

**Reply by robertsLando** [reply]
@hestiap what you could do is to enable sending zwave events to mqtt and subscribe to statistics updated event, that event is triggered by the driver every time there is an update of stats of a specific node (activity by the node). Another way if you wish to use a REST approach with mqtt you could use driverFunction api to retrieve just the lastActive prop of a specific node with a request

**Comment by hestiap** [comment]
@robertsLando 
I had again the issue with this motion sensor that was stuck
I have some log [zwavejs_2022-04-30.log](https://github.com/zwave-js/zwavejs2mqtt/files/8596835/zwavejs_2022-04-30.log)
Around 11 am there was a motion but nothing is z2m (time 2 hours before the time in the log)
Around 11:12 a refresh for the binary switch
Around 13:50 some reinterview, refresh... with no result
I saw an error in the log
`2022-04-30T11:52:21.991Z CNTRLR   [Node 099] Error during node interview: Node 99 does not support the Command Class Version!(ZW0302)`
Around 13:55 I cut the power and do a re-interview again, successful this time
I hope the log will help to find the issue...

**Reply by robertsLando** [reply]
@AlCalzone ?

**Reply by AlCalzone** [reply]
The device is contradicting itself. First it lists Version CC as supported:
``` 
2022-04-30T11:52:21.510Z CNTRLR « [Node 099] node info received
                                  supported CCs:
                                  · Z-Wave Plus Info
                                  · Firmware Update Meta Data
                                  · Version
                                  · Manufacturer Specific
                                  · Multi Channel
                                  · Device Reset Locally
                                  · Configuration
                                  · Powerlevel
                                  · Node Naming and Location
                                  · Notification
                                  · Association
                                  · Multi Channel Association
                                  · Association Group Information
                                  · Multilevel Sensor
                                  · Binary Switch
                                  · All Switch
                                  controlled CCs:
```
and when asked for the version of the Version CC, it claims not to support it anymore:
```
2022-04-30T11:52:21.985Z DRIVER « [Node 099] [REQ] [ApplicationCommand]
                                  └─[VersionCCCommandClassReport]
                                      CC:      Version
                                      version: 0
```

I added a workaround for this in v8.10.0 https://github.com/zwave-js/node-zwave-js/pull/4029
@hestiap Please upgrade, you're about 4 months behind ;)

**Comment by hestiap** [comment]
@AlCalzone 
thanks for your reply
I'm on the way to upgrade to zwavejs2mqtt: 6.8.1 / zwave-js: 9.0.7
I hope this is a good one
I've some troubles upgrading because I'm not so good with docker (not able to retrieve the settings, nodes from /var/lib/docker/volumes/zwavejs2mqtt/_data wich is where they should be found by zwavejs2mqtt). I will continue...

**Comment by hestiap** [comment]
@AlCalzone
I hope I did it right... zwavejs2mqtt: 6.8.1 / zwave-js: 9.0.7
=> `Error during node interview: Node #80 does not support the command  GetV1! (ZW0302)`
Not the same node ID but the same motion sensor
[zwavejs_2022-05-03.log](https://github.com/zwave-js/zwavejs2mqtt/files/8610278/zwavejs_2022-05-03.log)
![image](https://user-images.githubusercontent.com/73014953/166450063-c065c628-4eb7-4864-82d1-81c85d11671b.png)

**Reply by AlCalzone** [reply]
The node isn't responding to any commands. It acknowledges them but sends no response at all. 🤷‍♂️

**Reply by hestiap** [reply]
I've cut the power to restart the device, and the re-interview went to the end
I keep this version on my testing environment and then I'll upgrade the prod env around tomorrow 
I don't have really 2 env... but: 2 (RPI server + Controller) but only 1 zwave network
do you think I could have I side effects due to this?

**Reply by AlCalzone** [reply]
no that should be fine
